internal/events/pubsub/pubsub.go

```diff
@@ -0,0 +1,70 @@
+// Package pubsub contains utilities for handling Google Cloud Pub/Sub events.
+package pubsub
+
+import (
+	"fmt"
+	"regexp"
+
+	"cloud.google.com/go/functions/metadata"
+	"cloud.google.com/go/pubsub"
+	"github.com/GoogleCloudPlatform/functions-framework-go/internal/fftypes"
+)
+
+const (
+	pubsubEventType   = "google.pubsub.topic.publish"
+	pubsubMessageType = "type.googleapis.com/google.pubusb.v1.PubsubMessage"
+	pubsubService     = "pubsub.googleapis.com"
+)
+
+// LegacyEvent represents the legacy event payload that is sent by
+// Pub/Sub to Background Functions (https://cloud.google.com/functions/docs/writing/background).
+type LegacyEvent struct {
```

Comment:
I think `LegacyEvent` is a bit of an overloaded term because we also use it to distinguish between CloudEvents. Cloud we call this something else e.g. `RawPubSubEvent`

Reply:
I renamed this to "LegacyPushSubscriptionEvent" and made the comment more descriptive. I kept Legacy because that's how the [Cloud Functions docs ](https://cloud.google.com/functions/docs/calling/pubsub#legacy_cloud_pubsub_triggers) refer to them.

I didn't include PubSub in the struct name because golang has a naming convention to avoid "stuttering": https://blog.golang.org/package-names#TOC_3.